### DISCUSSION TITLE
Scaling and performance

### DISCUSSION THREAD

**Discussion by manfe513** [DeviceFarmer/stf#658] [Q&A]
Hi,can you please give us some advice on how to use marathon more effectively?

Our setup:

1. dedicated server with 72 cpu cores + 256Gb RAM, holds up to 72 emulators in docker containers(2 cpu core, 2048ram, 512 vm.heap)
2. teamcity runs a task:
- command line step runs within a docker container which has marathon v8 and open-stf client built-in
- we make open-stf connect and work with adb to remove and install our app on every emulator

In total, we have ~300 ui tests running within 11.5minutes
We tried 45 emulators with 4 cpu.cores and 4096ram - but the result is the same = 11minutes
We think it's because of context switching operations at the server CPU.

Maybe we can somehow improve our scheme/configs?Any help highly appreciated, thank you!

**Comment by jerimiah797** [comment]
As far as I can tell, those are fantastic results already…
